src/main/java/bio/terra/workspace/db/ControlledResourceDao.java

```diff
@@ -0,0 +1,83 @@
+package bio.terra.workspace.db;
+
+import bio.terra.workspace.service.controlledresource.exception.DuplicateControlledResourceException;
+import bio.terra.workspace.service.controlledresource.model.ControlledResourceMetadata;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ControlledResourceDao {
+  private final Logger logger = LoggerFactory.getLogger(ControlledResourceDao.class);
+  private final NamedParameterJdbcTemplate jdbcTemplate;
+
+  @Autowired
+  public ControlledResourceDao(NamedParameterJdbcTemplate jdbcTemplate) {
+    this.jdbcTemplate = jdbcTemplate;
+  }
+
+  public void createControlledResource(ControlledResourceMetadata controlledResource) {
+    final String sql =
+        "INSERT INTO workspace_resource "
+            + "(workspace_id, resource_id, associated_app, is_visible, owner, attributes) values "
+            + "(:workspaceId, :resourceId, :associatedApp, :isVisible, :owner, :attributes)";
+    final var params =
+        new MapSqlParameterSource()
+            .addValue("workspaceId", controlledResource.workspaceId())
+            .addValue("resourceId", controlledResource.resourceId())
+            .addValue("isVisible", controlledResource.isVisible())
+            .addValue("associatedApp", controlledResource.associatedApp().orElse(null))
+            .addValue("owner", controlledResource.owner().orElse(null))
+            .addValue("attributes", controlledResource.attributes().orElse(null));
+
+    try {
+      jdbcTemplate.update(sql, params);
+      logger.info(
+          String.format(
+              "Inserted record for workspaceResource %s",
+              controlledResource.resourceId().toString()));
+    } catch (DuplicateKeyException e) {
+      throw new DuplicateControlledResourceException(
+          String.format(
+              "Resource %s already exists in workspace %s",
+              controlledResource.resourceId(), controlledResource.workspaceId()),
+          e);
+    }
+  }
+
+  public Optional<ControlledResourceMetadata> getControlledResource(
+      UUID workspaceId, UUID resourceId) {
```

Comment:
Should I even take in the workspace id here if the resource id is sufficient?

Reply:
Might be redundant. Depends on what indexes are defined in the database.

Reply:
resource ID is the 
![Screen Shot 2021-02-11 at 9 49 05 AM](https://user-images.githubusercontent.com/53479492/107652298-702a4780-6c4e-11eb-9f11-86e0275d8616.png)
PK.